### DISCUSSION TITLE
Camera options (optical / thermal / wide / regular)

### DISCUSSION THREAD

**Discussion by mrbostad** [blakeblackshear/frigate#12759] [Ideas]
I would like to suggest the possibility to define camera type and lens type in the configuration, like 

camera_type: optical 
camera_type: thermal

and as a further suggestion, perhaps something like:

lens_type: wide
lens_type: regular
lens_type: narrow

(unifi doorbell has a little narrow lens, while my hikvision camera (optical) has wide, while the thermal part of the hikvision camera is regular.) 

that way perhaps the system can be even better to detect objects? since things can appear variously depending on camera type and lens type?

Just a thought. Thanks for a great product.

Greetings from proud Frigate+ user :)

**Comment by NickM-27** [comment]
Just to clarify, Images are cropped as regions so a wide vs standard vs narrow angle camera for example is not different at all from the perspective of object detection

**Reply by mrbostad** [reply]
> Just to clarify, Images are cropped as regions so a wide vs standard vs narrow angle camera for example is not different at all from the perspective of object detection

I agree on the theory of your comment, but i.ex. the Unifi Doorbell G4 camera, there is a lens distortion, and on the tapo C120 the image is "extra" wide, making things appear "narrower", so in example if i pass the tapo camera, i look slimmer and taller But on the Unifi G4 Bullet camera, things look "normal, the same on the reolink RLC-823A camera, that camera / lens also has some distortion.

**Reply by SleepinDevil** [reply]
@mrbostad I know I am dropping you this comment 1.5 years later but there are lens distortion filters natively supported on FFMPEG and you should apply those to any of your fisheye/wide angle lens cameras. I have done this this on the Reolink Doorbell and get perfectly square on images from it now which work really well for detections and as alerts on my HA

Just dropping you this message in case you didn't know about this feature :)
